relay-server/src/actors/store.rs

```diff
@@ -224,35 +83,69 @@ impl Producers {
     pub fn create(config: &Arc<Config>) -> Result<Self, ServerError> {
         let mut reused_producers = BTreeMap::new();
         let producers = Producers {
-            attachments: make_producer(&**config, &mut reused_producers, KafkaTopic::Attachments)?,
-            events: make_producer(&**config, &mut reused_producers, KafkaTopic::Events)?,
-            transactions: make_producer(
-                &**config,
+            attachments: Producer::create(
+                &config
+                    .kafka_config(KafkaTopic::Attachments)
+                    .map_err(|_| ServerErrorKind::KafkaError)?,
```

Comment:
I'm not sure how `make_producer`s `create_with_context` worked, but are we now erasing information from the original error that used to be preserved? Or was it always erased and mapped to a `ServerErrorKind::KafkaError`?

Reply:
The `create_with_context` call did not go anywhere, it  was moved into the kafka crate and still used but in `Producer::create` function - which captures and reports the errors. 

But we do erase the error here and replace it with `ServerErrorKind::KafkaError` which means that something wrong with kafka configuration and the producer could not be initialized. 

Do you think we wanna keep the error around here as well?  I can add required changes

Reply:
This is a nit by the way, but yes, I would give `KafkaError` a `#[cause] ProducerError` to propagate the information. Would that work?